cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -269,114 +285,690 @@ func copyFromGlobHelper(src, dst, globEx string) error {
 	return nil
 }
 
-func generateMountsModeRecover(mst *initramfsMountsState) error {
-	// steps 1 and 2 are shared with install mode
-	if err := generateMountsCommonInstallRecover(mst); err != nil {
-		return err
+// states for partition state
+const (
+	// states for UnlockState
+	partitionFound    = "found"
+	partitionNotFound = "not-found"
+	partitionErrFound = "err-finding"
+	// states for MountState
+	partitionErrMounting = "failed-to-mount"
+	partitionMounted     = "mounted"
+	// states for UnlockState
+	partitionUnlocked     = "unlocked"
+	partitionErrUnlocking = "failed-to-unlock"
+	// keys used to unlock for UnlockKey
+	keyRun      = "run"
+	keyFallback = "fallback"
+	keyRecovery = "recovery"
+)
+
+// partitionState is the state of a partition after recover mode has completed
+// for degraded mode.
+type partitionState struct {
+	// MountState is whether the partition was mounted successfully or not.
+	MountState string `json:"mount-state,omitempty"`
+	// MountLocation is where the partition was mounted.
+	MountLocation string `json:"mount-location,omitempty"`
+	// Device is what device the partition corresponds to.
+	Device string `json:"device,omitempty"`
+	// LocateState is whether the partitino on the disk was located or not.
+	LocateState string `json:"locate-state,omitempty"`
```

Comment:
should we s/locate/find/ ?

Reply:
renamed to FindState and `find-state` accordingly.